### DISCUSSION TITLE
Desktop track / route / waypoint manager

### DISCUSSION THREAD

**Discussion by caledavis** [devemux86/cruiser#495] [Q&A] [answered]
It would be excellent if cruiser had the ability to manage a track/route/waypoint library (as Basecamp did). It would allow the user to save these features, and edit them. Is this something that might be considered? 

I have been struggling to find an alternative now that Basecamp is effectively abandonware. Cruiser does an *excellent* job at the routing side of things, and this library management aspect is the only missing piece of the puzzle.

<img width="262" alt="Example" src="https://github.com/devemux86/cruiser/assets/74671557/48cc5194-f244-4332-95c5-5517be5516d5">

**Comment by devemux86** [comment, marked as answer]
Hi @caledavis,

Cruiser has a library for the favorites + routes + tracks.
(via the menu "My places")

You can save favorites + routes + tracks with different names, icons, colors, visibility and zoom level.
And select which favorites to use in navigation alerts.

<img src="https://user-images.githubusercontent.com/3484020/235928450-4fe2b991-2948-4630-b1cd-422093a47232.jpg" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/95152bf5-dac6-492d-a99e-4feae8d3c5ab" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/d0af3f6a-548e-4928-b181-5205730b47ab" width="33%"/>

**Reply by caledavis** [reply]
Thanks. Is this also available on the desktop version though? I haven’t been able to find it.

**Reply by devemux86** [reply]
@caledavis there are currently only available on Android.

**Reply by devemux86** [reply]
Also only Android Cruiser has the new:
- https://github.com/devemux86/cruiser/discussions/254

**Comment by MotoUKRider** [comment]
(this was going to be a simple comment in support of the request but has turned into something else!)

I am still using BaseCamp for this very reason - The filing system is pretty flexible (aside from not allowing duplicate Waypoint names), plus the offline (and speedy) routing make it easy to use on a Desktop or Laptop.  The (in)ability to deal with map themes is where I always found it lacking: The older "generic new" theme had the highway-shields for easy road identification, but wasn't easy to look at.  The Freizeitkarte ones have more colour (and elevation) but it isn't always easy to identify a road name without hovering over it.

My solution was to "port" my MotoUKRider theme (itself a version of "Biker") into a Garmin "style" and TYP file.  What a way to spend the hours (and coffee pot) :coffee:  I'm now at a position where the maps are much more pleasant, have the detail/colour I need, elevation and I can update them any time I like through a simple batch file (the joys if being an MS-DOS child).  The fact that it's such a pain (for the average user) to do anything creative with the map files in BaseCamp is part of the problem.

The map issues aside, it's hard to beat unless you need a cloud-based system.  Cruiser takes a Route GPX "as is" so I'm ready to ride, I multi-select my Waypoints and save those to a GPX, again - Cruiser will import those into a Favorite Group.  I currently just import everything as an orange "star" as it would be too much work to block-select "fuel", then "photo", then "hotel" to give each group its own Cruiser icon.

@caledavis if you think nice maps will help freshen the experience for you, I'll be happy to upload a small map (or extract) somewhere for you to try.  An extract of Calais is shown below, roads appear/get thicker as you zoom-in, with (blue) fuel icons appearing like in Cruiser with other POIs appearing at max-zoom.

![basecamp-calais-1](https://github.com/devemux86/cruiser/assets/125696109/715ba6a3-5a31-4033-a2d3-ed96ee0f80ff)

In Cruiser Desktop, I do find the ability to show all my Favorites (mass exported as a GPX from BaseCamp) useful if I were to plan a new route because all the interesting places are visible on the map - To do that in BaseCamp you need to put all your Waypoints in a List and start a new route from there (then copy/paste into a new Folder/list later.  Adding additional maps in the view is how it should be done, and with a simple text editor if you don't like map theme, it's easy to change and you *don't* need to recompile a 1.5GB UK map every time you tweak it.

So, in short(!), something in Desktop like an Outlook folder tree is something I'd welcome but I think the App has priority for @devemux86 so if it's just outdated/ugly mapping you dislike, there is a solution :smile:

**Reply by caledavis** [reply]
Thanks @MotoUKRider - I think we are in a similar position! I also batch exported from Basecamp every one of my GPX tracks, and now when planning a new route in Cruiser I import that as an overlay so I can see where I’ve been and decide which new places I’d like to explore. A library feature would be an excellent addition to the desktop version.

**Comment by 0709wiwiwi** [comment]
@caledavis @MotoUKRider.

> I currently import everything as an orange "star" because it would be too much work to block "fuel", then "photo", then "hotel" to give each group its own Cruiser icon.

A known gpx 1.1 (Y2004) imperfection is thus named.
The gpx sym tag makes little sense without associated icon.
Neither the gpx group nor Garmin have addressed the issue. 
Provided mutual agreement app devs can sure realise this.

References.

[Cruiser](https://github.com/devemux86/cruiser/discussions/127)
[Locus](https://help.locusmap.eu/topic/33948-transfer-any-image-as-a-usefull-thumbnail-map-icon)
[Organic maps](https://github.com/organicmaps/organicmaps/issues/2785#issuecomment-2018926826)
[OsmAnd](https://github.com/osmandapp/OsmAnd/issues/11285)
[Gpx group](https://groups.io/g/gpx/message/134)

**Reply by MotoUKRider** [reply]
@0709wiwiwi Yes, and I don't really mind that much.  I was tempted to write a Windoze app to read a "master" GPX from BaseCamp and then spew out as many GPX files as there were different icons (or named prefixes), but there's only so many riding hours in the day.

**Reply by MotoUKRider** [reply]
'fraid to say that I got impatient and wrote a Windoze app to process my "big" GPX of Waypoints as they come out of BaseCamp.

My naming conventions consist of `Food, Pizza Place` and `Photo, Nice View` etc, so with a series of "trigger" words I can export several GPX files with pre-defined colors/icons and anything left over gets put in the default "Waypoints" group.

![favorites-import-test](https://github.com/devemux86/cruiser/assets/125696109/68ceb284-971d-47fc-a550-eb53a076683a)

Although easy to (swipe) remove groups before (re)importing, I do find that if I ZIP my GPX files up and import "over the top" of the existing ones, I get a duplicate set of groups, with a "2" suffix.  These can even be the exact same files in the exact same ZIP - so there is no Name or Lat/Lon dupe check so far as I can see. (cc @devemux86 )

**Reply by devemux86** [reply]
@MotoUKRider nice colorful screenshots!
